### ISSUE TITLE
Add support for `init` option in docker-compose.yml

### ISSUE THREAD

**Issue by janeczku** [rancher/rancher#8608]
**Rancher Versions:**
Server: 1.5.4

Init option configures the container to run an init process as PID1 that forwards signals and reaps zombie processes. Docker run command now supports the `init`options. 
 ```
$ docker run --help | grep init
      --init                       Run an init inside the container (boolean flag)

 ```
See also v2.2 of the [docker-compose specification](https://github.com/docker/compose/blob/b38305f2d7b8dcb5d095cee8c57d908ac6011a9d/compose/config/config_schema_v2.2.json#L154).

Requirement:
We need to be able to specify those in either docker-compose or rancher-compose files.

**Comment by prysmakou** [comment]
We run Elasticsearch 5.3 images on Rancher. From time to time ES process becomes zombie. Elastic suggests to use '--init' flag which is not yet available in Rancher. It would be great to have docker-compose v2.2 support in Rancher.

**Comment by janeczku** [comment]
@prysmakou, thanks for the info!

**Comment by cjellick** [comment]
Needs the following changes:
- compose ( make sure round tripping in compose works, ie you see the values when you export config)
- cattle api (add to an existing integration test in test_docker.py)
- in the go-agent (can add a test here as well)(

**Comment by imikushin** [comment]
Since `--init-path` has been removed from `docker run` (see moby/moby#32355) and `--init` is a boolean, I'm going to add `init` as a boolean field.

**Comment by imikushin** [comment]
@cjellick Probably also go-rancher?

**Comment by cjellick** [comment]
@janeczku is your user specifically interested in init-path? Because docker has dropped that (code is ocmmited, not sure if its in a release yet) and we weren't going to implement it.

**Comment by janeczku** [comment]
@cjellick they were specifically interested in being able to set the init-path per container. The main reason being that they want to be able to pass arguments to their custom docker-init executable.

**Comment by joshwget** [comment]
@janeczku Couldn't an entrypoint just be used if that's the case?

**Comment by janeczku** [comment]
@joshwget sure, but the reason to use the new init docker run option is precisely so that you get all the benefits of an init process (zombie reaping, signal forwarding) without having to implement this in every of your images in the form of an entrypoint.

**Comment by janeczku** [comment]
@cjellick if that option is removed from docker engine API, there is really no way for us to implement a per container init-path option in rancher compose, right?

**Comment by joshwget** [comment]
If the `docker-init` binary is custom and set per container then having an `init-path` option isn't really any different than bind mounting that binary and setting an entrypoint. It wouldn't actually need to be packaged with each image if it's coming from the host.

**Comment by janeczku** [comment]
> If the docker-init binary is custom and set per container then having an init-path isn't really any different than bind mounting that binary and setting an entrypoint

There is a difference: The `--init` option passes both the container/image `entrypoint` and `cmd` to the init process. Your example would overwrite any existing entrypoint specified in the image and thus might break the container. Basically you would have to inspect the image for an existing entrypoint, then combine the init-path with the image entrypoint .

**Comment by StrongMonkey** [comment]
We need to cherry-pick https://github.com/rancher/agent/pull/172 into agent v1.6 then redo https://github.com/rancher/cattle/pull/2738 for cattle. Once these two are finished, update go-rancher and https://github.com/rancher/rancher-compose-executor/pull/54 should be good to go.

**Comment by deniseschannon** [comment]
Available in CLI v0.6.3-rc1 and rancher-compose-executor v0.14.11.

**Comment by deniseschannon** [comment]
Available in CLI v0.6.3-rc2 and rancher-compose-executor v0.14.12.

Note: `init` is still in the compose files, but the API field is `runInit`.

**Comment by sangeethah** [comment]
Tested with rancher-server build from v1.6 build.

1. Setting "runInit" to true in launch_configs for a service , results in containers getting deployed with init flag set to True.
2. Using the following yml , able to create services from UI which results in containers getting deployed with init flag set to True.
```
version: '2'
services:
  testme:
    image: sangeetha/testnewhostrouting
    init: true
```
3. Also able to use the above yml to create service with init set to true using rancher CLI - v0.6.3-rc2.

4. Exported config files for services created with init set to true , has init set to "true" in docker-compose.yml.
